src/test/java/org/openrewrite/java/migrate/lang/SwitchCaseAssigningToSwitchExpressionTest.java

```diff
@@ -0,0 +1,866 @@
+                  void assignment(int i) {
+                      int n = ( i = 2 );
+                      switch (i) {
+                          default: n = 5; break;
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void convertWhenOriginalVariableAssignmentIsComplexExpressionButNoSideEffects() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  String field = "strawberry";
+
+                  void doFormat(int i) {
+                      String orig = "initialValue" + "test" + 45 + true + field + this.field;
+                      switch (i) {
+                          default: orig = "hello"; break;
+                      }
+                  }
+              }
+              """,
+            """
+              class Test {
+                  String field = "strawberry";
+
+                  void doFormat(int i) {
+                      String orig = switch (i) {
+                          default: yield "hello";
+                      };
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void notConvertColonSwitchWithEmptyLastCase() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  enum TrafficLight {
+                      RED, GREEN, YELLOW
+                  }
+                  void doFormat(TrafficLight light) {
+                      String status = "initialValue";
+                      switch (light) {
+                          case RED: status = "stop"; break;
+                          case GREEN: status = "go"; break;
+                          case YELLOW:
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void notConvertSwitchOnUninitializedOriginalVariableAndNonExhaustiveSwitch() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  enum TrafficLight {
+                      RED, GREEN, YELLOW
+                  }
+                  void exhaustiveButCantAddDefaultAndMissingAssignment(TrafficLight light) {
+                      String status;
+                      switch (light) {
+                          case RED: status = "stop"; break;
+                          case GREEN: status = "go"; break;
+                          case YELLOW:
+                      }
+                  }
+
+                  void exhaustiveButCantAddDefaultAfterEmptyLabel(TrafficLight light) {
+                      String status = "initialValue";
+                      switch (light) {
+                          case RED: status = "stop"; break;
+                          case GREEN: status = "go"; break;
+                          case YELLOW:
+                      }
+                  }
```

Comment:
@timtebeek Are you sure this is correct?
While the switch is already exhaustive, wouldn't still adding `default: yield "initialValue";` at the end be safe and preserve the logic in any situation, given the last case is empty?
This would require a fix in the code but that new test made me wonder why not.